src/libcollections/str.rs

```diff
@@ -245,17 +264,19 @@ impl str {
         core_str::StrExt::as_bytes(self)
     }
 
-    /// Returns a raw pointer to the `&str`'s buffer.
+    /// Converts a string slice to a raw pointer.
     ///
-    /// The caller must ensure that the string outlives this pointer, and
-    /// that it is not
-    /// reallocated (e.g. by pushing to the string).
+    /// As string slices are a slice of bytes, the raw pointer points to a
+    /// `u8`. This pointer will be pointing to the first byte of the string
+    /// slice.
```

Comment:
Might be good idea to add a note that data pointed to by the pointer is not null-terminated.